openssl-sys/build.rs

```diff
@@ -52,7 +55,67 @@ fn main() {
     build_old_openssl_shim(&include_dirs);
 }
 
+macro_rules! import_options {
+    ( $( $name:ident $val:expr  )* ) => {
+       &[ $( (stringify!($name),$val), )* ]
+    };
+}
+
+fn generate_options_shim() -> PathBuf {
+    let options: &[(&'static str,u64)]=include!("src/ssl_options.rs");
+    let mut shim = String::new();
+    writeln!(shim,"#include <stdint.h>").unwrap();
+    writeln!(shim,"#include <openssl/ssl.h>").unwrap();
+
+    for &(name,value) in options {
+        writeln!(shim,"#define RUST_{} UINT64_C({})",name,value).unwrap();
+        writeln!(shim,"#ifndef {}",name).unwrap();
+        writeln!(shim,"# define {} 0",name).unwrap();
+        writeln!(shim,"#endif").unwrap();
+    }
+
+    writeln!(shim,"#define COPY_MASK ( \\").unwrap();
+
+    let mut it=options.iter().peekable();
+    while let Some(&(name,_))=it.next()  {
+        let eol=match it.peek() {
+            Some(_) => " | \\",
+            None    => " )"
+        };
+        writeln!(shim,"    ((RUST_{0}==(uint64_t)(uint32_t){0})?RUST_{0}:UINT64_C(0)){1}",name,eol).unwrap();
```

Comment:
I'm not super-familiar with C semantics, and I wanted to be sure it didn't do sign extension.